packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * useDebounce hook
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Use to debounce a value that is changing too frequently.
+ *
+ * Example usage:
+ *
+ *	// An example using the useDebounce hook with useResizeObserver()
+ *  // to debounce a rapidly changing width value
+ *
+ *	const [ resizeListener, sizes ] = useResizeObserver();
+ *	const debouncedSize = useDebounce(sizes.width, 100);
```

Comment:
Examples should adhere to coding standards.

```suggestion
 *	const debouncedSize = useDebounce( sizes.width, 100 );
```